src/librbd/ImageCtx.h

```diff
@@ -310,6 +311,7 @@ namespace librbd {
     void cancel_async_requests(Context *on_finish);
 
     void apply_metadata(const std::map<std::string, bufferlist> &meta);
+    void apply_qos();
```

Comment:
Why not add this method to ```ImageRequestWQ``` and invoke it directly? -- avoid the detour through ```ImageCtx```

Reply:
This method will be called when metadata updated.

Reply:
Yeah -- ```RefreshImage``` can invoke ```ImageRequestWQ``` methods

Reply:
But I think apply_qos() should be included in apply_metadata(). I mean qos metadatas is a part of image metadatas, so  I think it's better to call apply_qos() in apply_metadata().

Reply:
... again, not sure why you are so hesitent to have it invoke a method on ```ImageRequestWQ``` instead of adding another (unrelated) method to ```ImageCtx```

Reply:
Ha, got your point now. Yes, this should be in ImageRequestWQ. :+1: